models/demos/t3000/llama2_70b/tt/llama_rope.py

```diff
@@ -0,0 +1,143 @@
+import torch
+import ttnn
+from ttnn import ReplicateTensorToMesh, ConcatMeshToTensor
+from models.common.lightweightmodule import LightweightModule
+from models.demos.t3000.llama2_70b.tt.llama_common import precompute_freqs, get_rot_transformation_mat
+from loguru import logger
+
+
+def compute_gather_cos_sin(dhead, end, theta, position_ids, use_scaled_rope):
+    cos, sin = precompute_freqs(dhead, end, theta, use_scaled_rope)
+    position_id_expanded = position_ids.unsqueeze(1).expand(-1, cos.shape[-1])
+    cos = cos.gather(0, position_id_expanded)
+    sin = sin.gather(0, position_id_expanded)
+    cos = torch.stack([cos, cos], dim=-1).flatten(-2).unsqueeze(0).unsqueeze(0)
+    sin = torch.stack([sin, sin], dim=-1).flatten(-2).unsqueeze(0).unsqueeze(0)
```

Comment:
These lines look like a duplicate of llama_common::gather_cos_sin, can that be called here instead?